tests/pipelinesrun_test.go

```diff
@@ -0,0 +1,209 @@
+	pipelineName := "pipelines_run_int_test"
+	status, trigErr := testPipelinesRunService.TriggerPipelineRun(*PipelinesVcsBranch, pipelineName, isMultiBranch)
+	assert.NoError(t, trigErr)
+	assertTriggerRun(t, pipelineName, *PipelinesVcsBranch, status)
+
+	pollGetRunStatus(t, pipelineName)
+}
+
+func pollGetRunStatus(t *testing.T, pipelineName string) {
+	pollingAction := func() (shouldStop bool, responseBody []byte, err error) {
+		_, isMultiBranch, resourceErr := testPipelinesSyncService.GetPipelineResourceID(*PipelinesVcsRepoFullPath)
+		assert.NoError(t, resourceErr)
+		pipRunResponse, syncErr := testPipelinesRunService.GetRunStatus(*PipelinesVcsBranch, pipelineName, isMultiBranch)
+		assert.NoError(t, syncErr)
+
+		// Got the full valid response.
+		if pipRunResponse != nil && len(pipRunResponse.Pipelines) > 0 && pipRunResponse.Pipelines[0].Name == pipelineName {
+			log.Info("pipelines status code ", pipRunResponse.Pipelines[0].Run.StatusCode)
+			if isCancellable(pipRunResponse.Pipelines[0].Run.StatusCode) {
+
+				runStatusCode := pipRunResponse.Pipelines[0].Run.StatusCode
+				assertRunStatus(t, runStatusCode)
+				runID := pipRunResponse.Pipelines[0].Run.ID
+				run, cancelErr := testPipelinesRunService.CancelTheRun(runID)
+				assert.NoError(t, cancelErr)
+				assert.Equal(t, "cancelled run "+strconv.Itoa(runID)+" successfully", run)
+			} else {
+				return false, []byte{}, nil
+			}
+			return true, []byte{}, nil
+		}
+		return false, []byte{}, nil
+	}
+	pollingExecutor := &httputils.PollingExecutor{
+		Timeout:         defaultMaxWaitMinutes,
+		PollingInterval: defaultSyncSleepInterval,
+		PollingAction:   pollingAction,
+		MsgPrefix:       "Get pipeline run status...",
+	}
+	// polling execution
+	_, err := pollingExecutor.Execute()
+	assert.NoError(t, err)
+}
+
+func pollForSyncResourceStatus(t *testing.T) {
+	//define polling action
+	pollingAction := func() (shouldStop bool, responseBody []byte, err error) {
+		pipResStatus, body, syncErr := testPipelinesSyncStatusService.GetSyncPipelineResourceStatus(*PipelinesVcsRepoFullPath, *PipelinesVcsBranch)
+		assert.NoError(t, syncErr)
+
+		// Got the full valid response.
+		if len(pipResStatus) > 0 && pipResStatus[0].LastSyncStatusCode == 4002 {
+			return true, body, nil
+		}
+		return false, body, nil
+	}
+	pollingExecutor := &httputils.PollingExecutor{
+		Timeout:         defaultMaxWaitMinutes,
+		PollingInterval: defaultSyncSleepInterval,
+		PollingAction:   pollingAction,
+		MsgPrefix:       "Get pipeline sync status...",
+	}
+	// polling execution
+	_, err := pollingExecutor.Execute()
+	assert.NoError(t, err)
+}
+
+func isCancellable(statusCode int) bool {
+	switch statusCode {
+	case 4000:
+		fallthrough
+	case 4001:
+		fallthrough
+	case 4005:
+		fallthrough
+	case 4006:
+		fallthrough
+	case 4016:
+		fallthrough
+	case 4022:
+		return true
+
+	}
+	return false
+}
+
+func assertRunStatus(t *testing.T, statusCode int) {
+	assert.True(t, statusCode >= 4000 && statusCode <= 4022)
```

Comment:
Using GreaterOrEqual and LessOrEqual would print the status code in case of an error
```suggestion
	assert.GreaterOrEqual(t, statusCode, 4000)
	assert.LessOrEqual(t, statusCode, 4022)
```